### PR TITLE
net-im/psi: updated accroding to upstream changes

### PR DESCRIPTION
**Pull request by Ri0n**
Signed-off-by: Sergey Ilinykh rion4ik@gmail.com

Upstream is in progress of abandoning of qmake builds. So switch to cmake in advance.
Also added some flags for optional features.

### CODE REVIEW

File: net-im/psi/metadata.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 		<flag name='extras'>Enable Psi+ patcheset</flag>
```
Comment (a17r) [inline net-im/psi/metadata.xml]: please fix the typo

File: net-im/psi/metadata.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
...
 		<flag name='hunspell'>Enable spellchecking using <pkg>app-text/hunspell</pkg></flag>
 		<flag name='iconsets'>Install additional iconsets, some of them has not clear licensing</flag>
```
Comment (a17r) [inline net-im/psi/metadata.xml]: ..."some of them have no clear licensing"

File: net-im/psi/metadata.xml [hunk 3]
```diff
@@ -16,7 +16,7 @@
...
-		<flag name='sql'>Enable psi-new-history extras patch</flag>
+		<flag name='keyring'>Enable storing password in system keyring manager</flag>
 		<flag name='webengine'>Use <pkg>dev-qt/qtwebengine</pkg> instead of <pkg>dev-qt/qtwebkit</pkg></flag>
 		<flag name='webkit'>Enable themed, html-based chatlogs</flag>
```
Comment (a17r) [inline net-im/psi/metadata.xml]: Please use double quotes everywhere in this file.

File: net-im/psi/psi-9999.ebuild [hunk 4]
```diff
@@ -52,8 +51,9 @@ DEPEND="
 	x11-libs/libxcb
 	aspell? ( app-text/aspell )
 	dbus? ( dev-qt/qtdbus:5 )
-	enchant? ( >=app-text/enchant-1.3.0:0 )
+	enchant? ( app-text/enchant:* )
```
[the remark this answers is not in the document]
Comment (Ri0n) [reply net-im/psi/psi-9999.ebuild]: It's compatible. I'll just put 2 there.

File: net-im/psi/psi-9999.ebuild [hunk 5]
```diff
@@ -20,12 +20,11 @@ EGIT_MIN_CLONE_TYPE="single"
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS=""
-IUSE="aspell crypt dbus debug doc enchant extras +hunspell iconsets sql webengine webkit whiteboarding xscreensaver"
+IUSE="aspell crypt dbus debug doc enchant extras +hunspell iconsets keyring ssl webengine webkit webp whiteboarding xscreensaver"
```
[the remark this answers is not in the document]
Comment (Ri0n) [reply net-im/psi/psi-9999.ebuild]: cleaned up..

File: net-im/psi/psi-9999.ebuild [hunk 6]
```diff
@@ -52,8 +51,9 @@ DEPEND="
 	x11-libs/libxcb
 	aspell? ( app-text/aspell )
 	dbus? ( dev-qt/qtdbus:5 )
-	enchant? ( >=app-text/enchant-1.3.0:0 )
+	enchant? ( app-text/enchant:* )
 	hunspell? ( app-text/hunspell:= )
+	keyring? ( dev-libs/qtkeychain )
```
[the remark this answers is not in the document]
Comment (Ri0n) [reply net-im/psi/psi-9999.ebuild]: added :=. works

File: net-im/psi/psi-9999.ebuild [hunk 7]
```diff
@@ -113,16 +113,15 @@ src_unpack() {
...
 	if use extras; then
-		cp -a "${WORKDIR}/psi-plus/iconsets" "${S}" || die "failed to copy iconsets"
 		if use iconsets; then
 			cp -a "${WORKDIR}/resources/iconsets" "${S}" || die "failed to copy additional iconsets"
 		fi
 
-		eapply "${WORKDIR}/psi-plus/patches"/*.diff
-		use sql && eapply "${WORKDIR}/psi-plus/patches/dev/psi-new-history.patch"
+		patches=$(find "${WORKDIR}/psi-plus/patches" -maxdepth 1 -name '*.diff')
+		[ -n "$patches" ] && eapply "${WORKDIR}/psi-plus/patches"/*.diff
```
[the remark this answers is not in the document]
Comment (Ri0n) [reply net-im/psi/psi-9999.ebuild]: switched for Psi+ to snapshots report where all the patched applied already

File: net-im/psi/psi-9999.ebuild [hunk 8]
```diff
@@ -141,10 +140,12 @@ src_configure() {
...
 		$(use_enable hunspell)
+		$(use_enable keyring keychain)
 		$(use_enable xscreensaver xss)
 		$(use_enable whiteboarding)
 	)
 
+	use extras && CONF+=("--psiplus")
 	use debug && CONF+=("--debug")
 	use webengine && CONF+=("--enable-webkit" "--with-webkit=qtwebengine")
 	use webkit && CONF+=("--enable-webkit" "--with-webkit=qtwebkit")
```
[the remark this answers is not in the document]
Comment (Ri0n) [reply net-im/psi/psi-9999.ebuild]: switched to cmake.

File: net-im/psi/metadata.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ri0n) [reply net-im/psi/metadata.xml]: took your examples

File: net-im/psi/metadata.xml (hunk 1 above)
Comment (Ri0n) [reply net-im/psi/metadata.xml]: fixed

File: net-im/psi/metadata.xml (hunk 3 above)
Comment (Ri0n) [reply net-im/psi/metadata.xml]: fixed

File: net-im/psi/metadata.xml (hunk 2 above)
Comment (Ri0n) [reply net-im/psi/metadata.xml]: fixed

File: net-im/psi/metadata.xml [hunk 9]
```diff
@@ -9,17 +9,21 @@
 		<email>proxy-maint@gentoo.org</email>
 		<name>Proxy Maintainers</name>
 	</maintainer>
-	<longdescription>Psi is a very good XMPP client that uses Qt</longdescription>
+	<longdescription>Psi is a very good XMPP client that uses Qt and supporting wide range of XMPP
+		extensions. It's extremely configurable and extendable with plugins. It's supported on
+		almost every desktop system.
+	</longdescription>
```
Comment (a17r) [inline net-im/psi/metadata.xml]: this needs a bit improving, how about this:
``
Psi is a multi-platform XMPP client based on Qt and supports a wide range of XMPP
extensions. It is extremely configurable and extendable with plugins.
``

I can change on merge if you ack.

Comment (Ri0n) [reply net-im/psi/metadata.xml]: ack

**Comment by Ri0n** [comment]
With next PR I'll also add RDEPEND on psimedia.
Something like `voip? ( net-im/psimedia )`

**Comment by a17r** [comment]
> With next PR I'll also add RDEPEND on psimedia.
> Something like `voip? ( net-im/psimedia[extras?] )`

That should be considered carefully. Right now it would create a circular dependency, because net-im/psimeda already RDEPENDs on net-im/psi. Since optional RDEPENDs are somewhat frowned upon anyway (until EAPI-8 may add support for it properly) I would recommend an informational pkg_postinst message about net-im/psimedia instead.

**Comment by Ri0n** [comment]
> > With next PR I'll also add RDEPEND on psimedia.
> > Something like `voip? ( net-im/psimedia[extras?] )`
> 
> That should be considered carefully. Right now it would create a circular dependency, because net-im/psimeda already RDEPENDs on net-im/psi. Since optional RDEPENDs are somewhat frowned upon anyway (until EAPI-8 may add support for it properly) I would recommend an informational pkg_postinst message about net-im/psimedia instead.

oh. yes. circular... 
Well basically psimedia is a runtime only dependency (plugin) and I want it to be installed if a user want VoIP via use flags.
As far as I understand neither `rdepend` nor `pdepend` won't give me what I desire, and the easiest thing is to put it to `depend` in net-im/psi and remove `rdepend` on psi from `net-im/psimedia`

**Comment by Ri0n** [comment]
> Package is bundling, through git submodules, among others, an old version of `net-libs/http-parser` (2.8.1). Is it possible to unbundle this security sensitive library?

Let me check. I remember it was possible with qmake. Not sure about cmake.

**Comment by Ri0n** [comment]
> Let me check. I remember it was possible with qmake. Not sure about cmake.

Unfortunately unconditional. I'll poke the responsible person and I'll try to relink the module to something newer for now

**Comment by a17r** [comment]
> and the easiest thing is to put it to `depend` in net-im/psi

Which would be wrong as it is not a build relevant dependency. So either we reluctantly put it into RDEPEND or you decide that pkg_postinst is sufficient.

**Comment by Ri0n** [comment]
Let it be pkg_postinst with some message 
bold(flash(large_font("!!!!!! DON'T MISS VERY COOL FEATURE !!!!!!"))) :-D

**Comment by Ri0n** [comment]
bundled http-parser is bumped to 2.9.3

**Comment by a17r** [comment]
> bundled http-parser is bumped to 2.9.3

thanks, that was fast, preferably optional again in the future

**Comment by a17r** [comment]
having to manually fix your signoff, please fix it/use repoman or pkgcheck in the future.

**Comment by Ri0n** [comment]
Thanks,
Commiting via repoman is something impossible for me right now since I have Gentoo in chroot only and openpgp doesn't work in chroot correctly because of some incompatibility with kernel. So I have to commit without chroot and therefore without repoman.

**Pull request closed without merging**